### DISCUSSION TITLE
How to put actions-generated files into Pages site but not into repo itself?

### DISCUSSION THREAD

**Discussion by sidewayss** [community/community#144678] [Other Feature Feedback, Questions, & Ideas]
I have a build process that puts bundled/transpiled/minified files in a `dist/` directory.  I don't need that directory in the repo.  Those files are part of the release and NPM package, but they're not source.  On the other hand, I want them to be available online in my Pages site too.  How do I do that without making the `dist/` directory part of the repo itself?

This seems like a good idea to me, but I don't see anyone else doing it.  Is there reason why not?  Or is there an example out there? I am programming the build and release process for my small [html-elements](url) repo, if you want a reference point.

**Comment by JP-Dhabolt** [comment]
You want to set your Pages to be deployed via Github Actions, do your build, then upload the artifact and deploy using the provided actions.  See https://github.com/actions/upload-pages-artifact

**Comment by sidewayss** [comment]
@JP-Dhabolt - Thanks for the info and link. I'll give it a try.  I assume that the final step is to delete the `dist/` directory in that same Action.

**Reply by JP-Dhabolt** [reply]
If the `dist/` directory is being created in the Action itself, there is no need to delete, as it will disappear when the action workflow ends, since you aren't committing it.
